src/lib/utils/logger/logger.go

```diff
@@ -0,0 +1,64 @@
+package logger
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/user"
+	"time"
+)
+
+// FileLogger Logs into the secondary storage file
+var FileLogger *log.Logger
+
+// TerminalandFileLogger Logs into the secondary storage file and terminal
+var TerminalandFileLogger *log.Logger
+
```

Comment:
i think creating an interface for this should be better.

This should be a better way:

```
type Log struct {
  dir  string  // can be used to initialise the dir for logging. for now lets keep this explicitly as br-logs
  perm string // for permissions of opening or closing the file
}

func New() (*os.File, bool) { // this will do the work of SetupLogger()
  ...
}

func (lg Log) Terminal(message string, code string) { // code will be used to know whether the log is a normal
// message, a warning or a fatal.
  ....
}

func (lg Log) TFile(message string, code string) {
  ...
}

```

This should be the ideal way of creating any module. This will also minimize the long functional names as well and make maintaining easier.

Then, we can use this as,
```
import "github.com/zairza-cetb/..../logger"

log := logger.Log{
  dir: "br-logs",
  perm: "0755",
}

log.Terminal("this is a log", "n")

```

n -> normal
w -> warning
f -> fatal

Reply:
Some fatal logs, ask for an extra parameter, we will have to write all the cases,

Reply:
I thought of the same way, to create a different function and ask for codes for identifying the normal, fatal, warning logs, but it'd be simple if we let the developers use all the functionalities that log provides without having a layer on it to ask for codes.

Check the index here. https://golang.org/pkg/log/

Reply:
If they ask for extra parameters, then use the spread operator. The length of the spread operator can be used to differentiate between various types. If that doesnot work, using parameter as an `interface{}` is always an open option.

Reply:
I don't see any problem in keeping it raw, and moreover, are you sure you want to call the .New() function in every package, I think this is gonna create multiple log files. @Harkishen-Singh

Reply:
No. We will declare once inside the globalvars file. The reason I am saying is that then we can use this as a completely independent module and then we can publish this and would be useful for other projects or organisations.

Reply:
I am still having serious doubts regarding the design, specially to integrate all the functions that the raw package provides. Anyways, lets schedule a call whenever you are free, we will come to a conclusion and move forward.